smart-contracts/contracts-common/concordium-contracts-common/src/types.rs

```diff
@@ -798,10 +811,9 @@ impl str::FromStr for Timestamp {
 /// format in the UTC time zone.
 impl fmt::Display for Timestamp {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        use chrono::offset::TimeZone;
-        let time = self.timestamp_millis() as i64;
-        let date = chrono::Utc.timestamp_millis_opt(time).single().ok_or(fmt::Error)?;
-        write!(f, "{}", date.to_rfc3339())
+        let date: String = String::try_from(*self)
+            .map_err(|_| fmt::Error)?;
+        write!(f, "{}", date)
```

Comment:
I know this was incorrect before, but it is still incorrect since Display implementations are not meant to fail for "spurious" reasons
```
Additionally, the return value of this function is [fmt::Result](https://doc.rust-lang.org/std/fmt/type.Result.html) which is a type alias of [Result](https://doc.rust-lang.org/std/result/enum.Result.html)<(), [std::fmt::Error](https://doc.rust-lang.org/std/fmt/struct.Error.html)>. Formatting implementations should ensure that they propagate errors from the [Formatter](https://doc.rust-lang.org/std/fmt/struct.Formatter.html) (e.g., when calling [write!](https://doc.rust-lang.org/std/macro.write.html)). However, they should never return errors spuriously. That is, a formatting implementation must and may only return an error if the passed-in [Formatter](https://doc.rust-lang.org/std/fmt/struct.Formatter.html) returns an error. This is because, contrary to what the function signature might suggest, string formatting is an infallible operation. This function only returns a result because writing to the underlying stream might fail and it must provide a way to propagate the fact that an error has occurred back up the stack.
```

Reply:
Right so I can see three options: 
- Find library (or implement) conversion with to RFC3339 from `u64`.
- Deprecate/remove this function in favor of `String::try_from`.
- Instead of the error, return a string with something like `"Unsupported date representation"`. (Don't like this, but included it anyway)

Reply:
We discussed this in the office. 

Since we assume that JSON serialization never fails, and it would be annoying not to have a Display for timestamp, the solution we'll go with is that the Display implementation (and JSON implementation) will try to render it as a RFC3393 string, and if that fails fall back to just showing the timestamp.

The FromStr implementation will be updated to support both format as well.

Reply:
I also recommend not having the `TryFrom` for String implementation. It is too ambiguous.

We should instead add a method `to_rfc_3393` that can fail.

Reply:
I have updated it now.

I didn’t add `to_rfc_3393`. Looking at it, and thinking of it as a user, I would either was always like to end up with a string so IMHO `Display` is enough.